src/test/java/net/snowflake/client/jdbc/ConnectionIT.java

```diff
@@ -800,6 +803,36 @@ public void testStatementsAndResultSetsClosedByConnection() throws SQLException
     assertTrue(rs3.isClosed());
   }
 
+  @Test
+  public void testReadDateAfterSplittingResultSet() throws Exception {
+    Connection conn = getConnection();
+    try (Statement statement = conn.createStatement()) {
+      statement.execute("create or replace table table_with_date (int_c int, date_c date)");
+      statement.execute("insert into table_with_date values (1, '2015-10-25')");
+
+      try (ResultSet rs = statement.executeQuery("select * from table_with_date")) {
+        final SnowflakeResultSet resultSet = rs.unwrap(SnowflakeResultSet.class);
+        final List<SnowflakeResultSetSerializable> serializables =
+            resultSet.getResultSetSerializables(1 << 20);
```

Comment:
Perhaps can we avoid this bitshift operator? Generally we tend to use x*y notation, since it's more obvious. And we can extract it to final var named properly, so the reader doesn't wonder why this size has been chosen specifically.